src/game-loop/src/main-dude/states/MainDudeBaseState.cpp

```diff
@@ -0,0 +1,51 @@
+#include "main-dude/states/MainDudeBaseState.hpp"
+#include "main-dude/MainDude.hpp"
+#include "Input.hpp"
+
+void MainDudeBaseState::handle_movement(MainDude& main_dude, const Input& input, float speed)
+{
+    if (input.left())
+    {
+        main_dude._physics.add_velocity(-speed, 0.0f);
+    }
+    if (input.right())
+    {
+        main_dude._physics.add_velocity(speed, 0.0f);
+    }
+}
+
+MainDudeBaseState* MainDudeBaseState::handle_jumping(MainDude& main_dude, const Input& input)
+{
+    if (input.circle())
+    {
+        main_dude._physics.add_velocity(0.0f, -MainDude::JUMP_SPEED);
+        return &main_dude._states.jumping;
+    }
+    return nullptr;
+}
+
+MainDudeBaseState* MainDudeBaseState::handle_exiting(MainDude& main_dude, const Input& input)
```

Comment:
Why base state should be aware of something that concerns `MainDudeExitingState`?

Reply:
This abomination never happend